### ISSUE TITLE
[OpenShift/CRC] Can't get kuma-demo running

### ISSUE THREAD

**Issue by devops-42** [kumahq/kuma#2174]
### Summary

Hi,

To play around with kuma I tried to install it on OpenShift (CRC). The installation of the control plane succeeded. When deploying the kuma-demo all pods stuck in `ContainerCreating`, the logs shows:

```
(combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kuma-demo-app-6787b4f7f5-5476c_kuma-demo_839e4fe7-1bac-4aac-ab54-023008325a5f_0(363a03f9e98efcc6517b49634f0f87750c686abe9ce6da25daeaa3e7f4412e2c): [kuma-demo/kuma-demo-app-6787b4f7f5-5476c:kuma-cni]: error adding container to network "kuma-cni": exit status 1
```

How to get rid of this error?

Thanks for your help in advance!
Cheers

### Steps To Reproduce

1. Install and start CRC
2. Follow the instructions on installiing kuma:
```
kumactl install control-plane --cni-enabled | oc apply -f -
oc adm policy add-scc-to-group anyuid system:serviceaccounts:kuma-demo
```
3. Create a route to the `kuma-control-plane` service and map port 5681
4. Verify, whether GUI is working
5. Try to install kuma-demo:
```
oc apply -f https://bit.ly/demokuma
```

4 pods will be created, all stuck with the error above. 

### Additional Details & Logs

- Version: Kuma v1.1.6 on Kubernetes
- Error logs: Message see above
- Platform and Operating System: CRC 1.26.0 on MacOS
- Installation Method: kumactl

**Comment by bartsmykla** [comment]
I don't unfortunately have access to OpenShift, but could you see if there is anything interesting in `kuma-cni-node` `DaemonSet` in `kuma-system` namespace?

**Comment by bartsmykla** [comment]
I mean, could you look at the logs and see if there is anything what could explain this error?

**Comment by devops-42** [comment]
pardon, I forgot, here we go:

```
Wrote kuma CNI binaries to /host/opt/cni/bin.
/host/secondary-bin-dir is non-writable, skipping
Using CNI config template from CNI_NETWORK_CONFIG environment variable.
CNI config: {
  "cniVersion": "0.3.1",
  "name": "kuma-cni",
  "type": "kuma-cni",
  "log_level": "info",
  "kubernetes": {
      "kubeconfig": "/etc/cni/multus/net.d/ZZZ-kuma-cni-kubeconfig",
      "cni_bin_dir": "/var/lib/cni/bin",
      "exclude_namespaces": [ "kube-system" ]
  }
}
Created CNI config kuma-cni.conf
Done configuring CNI.  Sleep=true
Terminated
Cleaning up and exiting.
Removing kuma-cni net.d conf file: /host/etc/cni/net.d/kuma-cni.conf
Removing kuma-cni kubeconfig file: /host/etc/cni/net.d/ZZZ-kuma-cni-kubeconfig
Removing existing binaries
Exiting.
Wrote kuma CNI binaries to /host/opt/cni/bin.
/host/secondary-bin-dir is non-writable, skipping
Using CNI config template from CNI_NETWORK_CONFIG environment variable.
CNI config: {
  "cniVersion": "0.3.1",
  "name": "kuma-cni",
  "type": "kuma-cni",
  "log_level": "info",
  "kubernetes": {
      "kubeconfig": "/etc/cni/multus/net.d/ZZZ-kuma-cni-kubeconfig",
      "cni_bin_dir": "/var/lib/cni/bin",
      "exclude_namespaces": [ "kube-system" ]
  }
}
Created CNI config kuma-cni.conf
Done configuring CNI.  Sleep=true
```

**Comment by bartsmykla** [comment]
Ok, nothing here. Are you able to get the logs from `kubelet`?

**Comment by devops-42** [comment]
Sure, but there's a plethora of log lines when grepping for ´kuma`. I have extracted the hopefully relevant part:
```
Jun 16 12:42:01 crc-gm7cm-master-0 hyperkube[2881]: I0616 12:42:01.749877    2881 kuberuntime_manager.go:443] No sandbox for pod "kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo(8fd222b9-dafc-4138-b668-368f826a85ab)" can be found. Need to start a new one
Jun 16 12:42:08 crc-gm7cm-master-0 hyperkube[2881]: I0616 12:42:08.749745    2881 kuberuntime_manager.go:443] No sandbox for pod "kuma-demo-app-6787b4f7f5-5476c_kuma-demo(839e4fe7-1bac-4aac-ab54-023008325a5f)" can be found. Need to start a new one
Jun 16 12:42:09 crc-gm7cm-master-0 hyperkube[2881]: I0616 12:42:09.751788    2881 kuberuntime_manager.go:443] No sandbox for pod "redis-master-55fd8f6f54-lfkb7_kuma-demo(f83b0855-f837-4528-b02e-5a0743f54b10)" can be found. Need to start a new one
Jun 16 12:42:10 crc-gm7cm-master-0 hyperkube[2881]: I0616 12:42:10.750148    2881 kuberuntime_manager.go:443] No sandbox for pod "postgres-master-645bc44fd-dtjch_kuma-demo(639fdddb-6e03-4e82-bf6b-67362fc56f49)" can be found. Need to start a new one
Jun 16 12:42:14 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:14.219672    2881 remote_runtime.go:116] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo_8fd222b9-dafc-4138-b668-368f826a85ab_0(8524e5698177d104e785f774d22f65e6bb0e1613afc1d41e41cbef6f2f8cdba6): [kuma-demo/kuma-demo-backend-v0-56db47c579-mr8dv:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:14 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:14.219781    2881 kuberuntime_sandbox.go:70] CreatePodSandbox for pod "kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo(8fd222b9-dafc-4138-b668-368f826a85ab)" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo_8fd222b9-dafc-4138-b668-368f826a85ab_0(8524e5698177d104e785f774d22f65e6bb0e1613afc1d41e41cbef6f2f8cdba6): [kuma-demo/kuma-demo-backend-v0-56db47c579-mr8dv:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:14 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:14.219812    2881 kuberuntime_manager.go:767] createPodSandbox for pod "kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo(8fd222b9-dafc-4138-b668-368f826a85ab)" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo_8fd222b9-dafc-4138-b668-368f826a85ab_0(8524e5698177d104e785f774d22f65e6bb0e1613afc1d41e41cbef6f2f8cdba6): [kuma-demo/kuma-demo-backend-v0-56db47c579-mr8dv:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:14 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:14.219912    2881 pod_workers.go:191] Error syncing pod 8fd222b9-dafc-4138-b668-368f826a85ab ("kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo(8fd222b9-dafc-4138-b668-368f826a85ab)"), skipping: failed to "CreatePodSandbox" for "kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo(8fd222b9-dafc-4138-b668-368f826a85ab)" with CreatePodSandboxError: "CreatePodSandbox for pod \"kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo(8fd222b9-dafc-4138-b668-368f826a85ab)\" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kuma-demo-backend-v0-56db47c579-mr8dv_kuma-demo_8fd222b9-dafc-4138-b668-368f826a85ab_0(8524e5698177d104e785f774d22f65e6bb0e1613afc1d41e41cbef6f2f8cdba6): [kuma-demo/kuma-demo-backend-v0-56db47c579-mr8dv:kuma-cni]: error adding container to network \"kuma-cni\": exit status 1"
Jun 16 12:42:21 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:21.657835    2881 remote_runtime.go:116] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kuma-demo-app-6787b4f7f5-5476c_kuma-demo_839e4fe7-1bac-4aac-ab54-023008325a5f_0(d1350ea5f543422bd95308cd5f4c6fec8760970bb85c0271d488265c20081262): [kuma-demo/kuma-demo-app-6787b4f7f5-5476c:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:21 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:21.657884    2881 kuberuntime_sandbox.go:70] CreatePodSandbox for pod "kuma-demo-app-6787b4f7f5-5476c_kuma-demo(839e4fe7-1bac-4aac-ab54-023008325a5f)" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kuma-demo-app-6787b4f7f5-5476c_kuma-demo_839e4fe7-1bac-4aac-ab54-023008325a5f_0(d1350ea5f543422bd95308cd5f4c6fec8760970bb85c0271d488265c20081262): [kuma-demo/kuma-demo-app-6787b4f7f5-5476c:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:21 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:21.657902    2881 kuberuntime_manager.go:767] createPodSandbox for pod "kuma-demo-app-6787b4f7f5-5476c_kuma-demo(839e4fe7-1bac-4aac-ab54-023008325a5f)" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kuma-demo-app-6787b4f7f5-5476c_kuma-demo_839e4fe7-1bac-4aac-ab54-023008325a5f_0(d1350ea5f543422bd95308cd5f4c6fec8760970bb85c0271d488265c20081262): [kuma-demo/kuma-demo-app-6787b4f7f5-5476c:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:21 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:21.657980    2881 pod_workers.go:191] Error syncing pod 839e4fe7-1bac-4aac-ab54-023008325a5f ("kuma-demo-app-6787b4f7f5-5476c_kuma-demo(839e4fe7-1bac-4aac-ab54-023008325a5f)"), skipping: failed to "CreatePodSandbox" for "kuma-demo-app-6787b4f7f5-5476c_kuma-demo(839e4fe7-1bac-4aac-ab54-023008325a5f)" with CreatePodSandboxError: "CreatePodSandbox for pod \"kuma-demo-app-6787b4f7f5-5476c_kuma-demo(839e4fe7-1bac-4aac-ab54-023008325a5f)\" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_kuma-demo-app-6787b4f7f5-5476c_kuma-demo_839e4fe7-1bac-4aac-ab54-023008325a5f_0(d1350ea5f543422bd95308cd5f4c6fec8760970bb85c0271d488265c20081262): [kuma-demo/kuma-demo-app-6787b4f7f5-5476c:kuma-cni]: error adding container to network \"kuma-cni\": exit status 1"
Jun 16 12:42:22 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:22.221324    2881 remote_runtime.go:116] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_redis-master-55fd8f6f54-lfkb7_kuma-demo_f83b0855-f837-4528-b02e-5a0743f54b10_0(9564585baddfdd041204fb48d9aba6c9e6661c568425d5e0ed9712f890b71379): [kuma-demo/redis-master-55fd8f6f54-lfkb7:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:22 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:22.221447    2881 kuberuntime_sandbox.go:70] CreatePodSandbox for pod "redis-master-55fd8f6f54-lfkb7_kuma-demo(f83b0855-f837-4528-b02e-5a0743f54b10)" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_redis-master-55fd8f6f54-lfkb7_kuma-demo_f83b0855-f837-4528-b02e-5a0743f54b10_0(9564585baddfdd041204fb48d9aba6c9e6661c568425d5e0ed9712f890b71379): [kuma-demo/redis-master-55fd8f6f54-lfkb7:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:22 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:22.221474    2881 kuberuntime_manager.go:767] createPodSandbox for pod "redis-master-55fd8f6f54-lfkb7_kuma-demo(f83b0855-f837-4528-b02e-5a0743f54b10)" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_redis-master-55fd8f6f54-lfkb7_kuma-demo_f83b0855-f837-4528-b02e-5a0743f54b10_0(9564585baddfdd041204fb48d9aba6c9e6661c568425d5e0ed9712f890b71379): [kuma-demo/redis-master-55fd8f6f54-lfkb7:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:22 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:22.221514    2881 pod_workers.go:191] Error syncing pod f83b0855-f837-4528-b02e-5a0743f54b10 ("redis-master-55fd8f6f54-lfkb7_kuma-demo(f83b0855-f837-4528-b02e-5a0743f54b10)"), skipping: failed to "CreatePodSandbox" for "redis-master-55fd8f6f54-lfkb7_kuma-demo(f83b0855-f837-4528-b02e-5a0743f54b10)" with CreatePodSandboxError: "CreatePodSandbox for pod \"redis-master-55fd8f6f54-lfkb7_kuma-demo(f83b0855-f837-4528-b02e-5a0743f54b10)\" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_redis-master-55fd8f6f54-lfkb7_kuma-demo_f83b0855-f837-4528-b02e-5a0743f54b10_0(9564585baddfdd041204fb48d9aba6c9e6661c568425d5e0ed9712f890b71379): [kuma-demo/redis-master-55fd8f6f54-lfkb7:kuma-cni]: error adding container to network \"kuma-cni\": exit status 1"
Jun 16 12:42:22 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:22.707064    2881 remote_runtime.go:116] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_postgres-master-645bc44fd-dtjch_kuma-demo_639fdddb-6e03-4e82-bf6b-67362fc56f49_0(3089647d9982cb3b1bcf13567449c26eee572deef25cfa9ce8234f81ad9f8337): [kuma-demo/postgres-master-645bc44fd-dtjch:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:22 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:22.707121    2881 kuberuntime_sandbox.go:70] CreatePodSandbox for pod "postgres-master-645bc44fd-dtjch_kuma-demo(639fdddb-6e03-4e82-bf6b-67362fc56f49)" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_postgres-master-645bc44fd-dtjch_kuma-demo_639fdddb-6e03-4e82-bf6b-67362fc56f49_0(3089647d9982cb3b1bcf13567449c26eee572deef25cfa9ce8234f81ad9f8337): [kuma-demo/postgres-master-645bc44fd-dtjch:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:22 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:22.707139    2881 kuberuntime_manager.go:767] createPodSandbox for pod "postgres-master-645bc44fd-dtjch_kuma-demo(639fdddb-6e03-4e82-bf6b-67362fc56f49)" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_postgres-master-645bc44fd-dtjch_kuma-demo_639fdddb-6e03-4e82-bf6b-67362fc56f49_0(3089647d9982cb3b1bcf13567449c26eee572deef25cfa9ce8234f81ad9f8337): [kuma-demo/postgres-master-645bc44fd-dtjch:kuma-cni]: error adding container to network "kuma-cni": exit status 1
Jun 16 12:42:22 crc-gm7cm-master-0 hyperkube[2881]: E0616 12:42:22.707269    2881 pod_workers.go:191] Error syncing pod 639fdddb-6e03-4e82-bf6b-67362fc56f49 ("postgres-master-645bc44fd-dtjch_kuma-demo(639fdddb-6e03-4e82-bf6b-67362fc56f49)"), skipping: failed to "CreatePodSandbox" for "postgres-master-645bc44fd-dtjch_kuma-demo(639fdddb-6e03-4e82-bf6b-67362fc56f49)" with CreatePodSandboxError: "CreatePodSandbox for pod \"postgres-master-645bc44fd-dtjch_kuma-demo(639fdddb-6e03-4e82-bf6b-67362fc56f49)\" failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_postgres-master-645bc44fd-dtjch_kuma-demo_639fdddb-6e03-4e82-bf6b-67362fc56f49_0(3089647d9982cb3b1bcf13567449c26eee572deef25cfa9ce8234f81ad9f8337): [kuma-demo/postgres-master-645bc44fd-dtjch:kuma-cni]: error adding container to network \"kuma-cni\": exit status 1"
```

**Comment by dmilde** [comment]
Had a similar error with permission issues on Openshift 4.6.X with kuma 1.1.6.
It was fixed and is now working for me with the new v1.2.0, maybe give it a try?

**Comment by devops-42** [comment]
@dmilde Thanks for the hint, with 1.2.0 I could install the demo application. For installing the `kuma-metrics` I had to add the following scc command:
```
oc adm policy add-scc-to-group anyuid system:serviceaccounts:kuma-metrics
```
